src/analysis/seminorm.lean

```diff
@@ -255,26 +256,50 @@ variables [add_monoid E]
 section has_scalar
 variables [has_scalar 𝕜 E]
 
-instance : inhabited (seminorm 𝕜 E) :=
-⟨{ to_fun    := λ _, 0,
-   smul'     := λ _ _, (mul_zero _).symm,
-   triangle' := λ x y, by rw add_zero }⟩
+instance fun_like : fun_like (seminorm 𝕜 E) E (λ _, ℝ) :=
+{ coe := seminorm.to_fun, coe_injective' := λ f g h, by cases f; cases g; congr' }
 
+/-- Helper instance for when there's too many metavariables to apply `to_fun.to_coe_fn`. -/
 instance : has_coe_to_fun (seminorm 𝕜 E) (λ _, E → ℝ) := ⟨λ p, p.to_fun⟩
 
-@[ext] lemma ext {p q : seminorm 𝕜 E} (h : (p : E → ℝ) = q) : p = q :=
-begin
-  cases p,
-  cases q,
-  have : p_to_fun = q_to_fun := h,
-  simp_rw this,
-end
+@[ext] lemma ext {p q : seminorm 𝕜 E} (h : ∀ x, (p : E → ℝ) x = q x) : p = q := fun_like.ext p q h
+
+instance : has_zero (seminorm 𝕜 E) :=
+⟨{ to_fun    := 0,
+  smul'     := λ _ _, (mul_zero _).symm,
+  triangle' := λ _ _, by rw add_zero }⟩
```

Comment:
My guess is that this will work (untested)
```suggestion
  triangle' := λ _ _, (add_zero _).ge }⟩
```

Reply:
oh, I didn't see that (came up with the same thing myself)